src/utils.ts

```diff
@@ -1,20 +1,34 @@
-import * as https from 'https';
+import axiosBase, { AxiosRequestConfig } from "axios";
+const ProxyAgent = require("proxy-agent");
 
-export function httpsRequest(url: string, options: https.RequestOptions = {}, encoding = 'utf8'): Promise<string> {
-	return new Promise((resolve, reject) => {
-		https.request(url, options, res => {
-			if (res.statusCode === 301 || res.statusCode === 302) { // follow redirects
-				return resolve(httpsRequest(res.headers.location, options, encoding));
-			}
-			if (res.statusCode !== 200) {
-				return reject(res.statusMessage);
-			}
-			let body = '';
-			res.setEncoding(encoding)
-				.on('data', data => body += data)
-				.on('end', () => resolve(body));
-		})
-			.on('error', reject)
-			.end();
-	});
+const httpProxy = process.env["HTTP_PROXY"] || process.env["http_proxy"];
+const httpsProxy = process.env["HTTPS_PROXY"] || process.env["https_proxy"];
+
+let proxyConf = {};
+
+if (httpProxy || httpsProxy) {
+  console.log(`Found proxy setting. Using to download files.`);
+  console.log(`HTTP: ${httpProxy}`);
+  console.log(`HTTPS: ${httpProxy}`);
+  proxyConf = {
+    proxy: false,
+    httpAgent: httpProxy ? new ProxyAgent(httpProxy) : undefined,
+    httpsAgent: httpsProxy ? new ProxyAgent(httpsProxy) : undefined,
+  };
+}
+
+const axios = axiosBase.create({ ...proxyConf });
+
+export async function request(
+  url: string,
+  options: AxiosRequestConfig = {}
+): Promise<any> {
+  console.log(`Request to: ${url} with options: ${JSON.stringify(options)}`);
+  try {
+    const result = await axios.get(url, { ...options });
+    return result.data;
+  } catch (e: any) {
+    console.error(e.toJSON());
+    throw new Error(`Request for terraform binary failed: ${e && e.message}`);
```

Comment:
Going to ask internally about the usage for this library, but I think we should follow the previous pattern of the resolve/reject with promises instead of try/catch and printing a message to the console.

Reply:
Reviewed with the team and we agree the approach here is good with the exception of catching the error here. We shouldn't print the error to the console and throw a new formatted here.

This library is used for more than downloading terraform, so instead we should let the caller handle the exception.

Reply:
Remove the try catch so the caller must handle the error